src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -3311,6 +3311,22 @@ namespace winrt::TerminalApp::implementation
 
             const IInspectable unused{ nullptr };
             _SetAsDefaultDismissHandler(unused, unused);
+
+            // TEMPORARY SOLUTION
+            // If the connection has requested for the window to be maximized,
+            // manually maximize it here. Ideally, we should be _initializing_
+            // the session maximized, instead of manually maximizing it after initialization.
+            // However, because of the current way our defterm handoff works,
+            // we are unable to get the connection info before the terminal session
+            // has already started.
+
+            // Make sure that there were no other tabs already existing (in
+            // the case that we are in glomming mode), because we don't want
+            // to be maximizing other existing sessions that did not ask for it.
+            if (_tabs.Size() == 1 && connection.ShowWindow() == 3)
```

Comment:
nit: `connection.ShowWindow() == 3` doesn't really mean much to me. Maybe create a `static const` variable in the line above for what that 3 means?

Reply:
I think this should be SW_MAXIMIZE, right?

Where is STARTF_USESHOWWINDOW checked?

Reply:
turns out `SW_SHOWMAXIMIZED` was already defined! Fixed it

Reply:
> Where is STARTF_USESHOWWINDOW checked?

Here, we don't set the `_wShowWindow` flag unless `STARTF_USESHOWWINDOW` is set: https://github.com/microsoft/terminal/blob/main/src/host/settings.cpp#:~:text=if%20(WI_IsFlagSet(dwFlags%2C%20STARTF_USESHOWWINDOW,%7D

Reply:
> Here, we don't set the _wShowWindow flag unless STARTF_USESHOWWINDOW is set:

for what it's worth, it is initialized to zero (`SW_HIDE`), and in the future when we want to support it we will not be able to do so because we cannot discern `0 & STARTF_USESHOWWINDOW` (so, `SW_HIDE`) from `0`